src/test_workflow/test_cluster.py

```diff
@@ -83,16 +83,16 @@ def start(self) -> None:
     def terminate(self) -> None:
         if self.service:
             self.termination_result = self.service.terminate()
+            self.__save_test_result_data(self.termination_result)
 
         for service in self.dependencies:
             termination_result = service.terminate()
             self.__save_test_result_data(termination_result)
+            service.uninstall()
 
         if not self.termination_result:
             raise ClusterServiceNotInitializedException()
 
-        self.__save_test_result_data(self.termination_result)
-
         self.service.uninstall()
```

Comment:
Could you move this to the `if self.service` as well so uninstall is handled in the correct order?
Thanks.